.buildkite/pipeline_generator/pipeline_generator.py

```diff
@@ -0,0 +1,180 @@
+import yaml
+import click
+from typing import List, Dict, Union
+import os
+
+from plugin import (
+    get_kubernetes_plugin_config,
+    get_docker_plugin_config,
+)
+from utils import (
+    AgentQueue,
+    AMD_REPO,
+    TEST_PATH,
+    EXTERNAL_HARDWARE_TEST_PATH,
+    PIPELINE_FILE_PATH,
+    STEPS_TO_BLOCK,
+    get_agent_queue,
+    get_full_test_command,
+    get_image_path,
+    get_multi_node_test_command,
+)
+from step import (
+    TestStep, 
+    BuildkiteStep, 
+    BuildkiteBlockStep, 
+    get_block_step, 
+    get_step_key
+)
+
+
+def _get_plugin_config(step: TestStep) -> Dict:
+    """
+    Returns the plugin configuration for the step.
+    If the step is run on A100 GPU, use k8s plugin since A100 node is on k8s.
+    Otherwise, use Docker plugin.
+    """
+    test_bash_command = [
+        "bash", 
+        "-c", 
+        get_full_test_command(step.commands, step.working_dir)
+    ]
+    docker_image_path = get_image_path()
+
+    if step.gpu == "a100":
+        return get_kubernetes_plugin_config(
+            docker_image_path, 
+            test_bash_command
+        )
+    return get_docker_plugin_config(
+        docker_image_path, 
+        test_bash_command, 
+        step.no_gpu
+    )
+
+def read_test_steps(file_path: str) -> List[TestStep]:
+    """Read test steps from test pipeline yaml and parse them into Step objects."""
+    with open(file_path, "r") as f:
+        content = yaml.safe_load(f)
+    return [TestStep(**step) for step in content["steps"]]
+
+def process_step(step: TestStep, run_all: str, list_file_diff: List[str]) -> List[Union[BuildkiteStep, BuildkiteBlockStep]]:
```

Comment:
feels that `run_all` should be a `bool`?

Reply:
yes, but the shell script that calls this script is sending it as a string